### ISSUE TITLE
[Discussion] Potential enhancements.

### ISSUE THREAD

**Issue by e0lithic** [Mister-Hope/vuepress-theme-hope#172]
Proposed Enhancements

- [x] Add switching animation when transiting from guide to homepage as well

- [x] ~~Possible way to configure code fences to use dark colors in global light theme.~~

- [x] Configurable font size and typography for better utilization of space, for eg : https://nuxtjs.org/guide

- [x] Alternative logo in the navigation bar for dark mode.

- [x] The default search could be extended using flexsearch to include full text search without the need for algolia for private documentation as is being done by [vuepress-plugin-fulltext-search](https://github.com/leo-buneev/vuepress-plugin-fulltext-search)

**Comment by Mister-Hope** [comment]
> Alternative logo in the navigation bar for dark mode.

Nice catch, I will add an option

**Comment by Mister-Hope** [comment]
The first one is added in [bf4a8c](https://github.com/Mister-Hope/vuepress-theme-hope/commit/bf4a8c28512736ccb1060510f2350bb46dae9ae6)

Thanks again

**Comment by e0lithic** [comment]
The default search could be extended using flexsearch include full text search without the need for algolia for private documentation as is being done by [vuepress-plugin-fulltext-search](https://github.com/leo-buneev/vuepress-plugin-fulltext-search)

**Comment by Mister-Hope** [comment]
> The default search could be extended using flexsearch to include full text search without the need for algolia for private documentation as is being done by [vuepress-plugin-fulltext-search](https://github.com/leo-buneev/vuepress-plugin-fulltext-search)

Thanks for the advice, I will check it out carefully later.

I heard fulltext search, it seems to search through a user(or plugin) created search base. So I am not sure if this vuepress plugin can work well on a huge mount of docs. But I will have a try.

**Comment by Mister-Hope** [comment]
>  Possible way to configure code fences to use dark colors in global light theme.

I won't build this feature, instead, I will try to building a bigger one. I am looking for a way to let the users config the code fence theme. But I found it's a bit hard unless I edit vuepress files.

**Comment by e0lithic** [comment]
@Mister-Hope I understand that for the breadcrumbs to work readme.md is required. However, it seems that the heirarchy path is doesn't take the folder and rather takes the readme.md as the parent heirarchy. You have already mentioned in the doc, but is there a workaround for it ?

**Comment by Mister-Hope** [comment]
currently the work around is disabling it😂I will make an enhance later.

**Comment by Mister-Hope** [comment]
> @Mister-Hope I understand that for the breadcrumbs to work readme.md is required. However, it seems that the heirarchy path is doesn't take the folder and rather takes the readme.md as the parent heirarchy. You have already mentioned in the doc, but is there a workaround for it ?

I am not sure what do you mean, cound you explain it with more detail? It seems that it has grammar issues, and I am not sure if I understand it correctlly.

**Comment by e0lithic** [comment]
Taking the example from the image below. Here is a demo example
```
.
├── guide
│   ├── basic-config.md
│   ├── directory-structure.md
│   ├── install.md
│   └── README.md
└ ......
```
Here the file README.md contains the `Introduction`. The file directory-structure.md is in parallel and technically does not reside inside README.md. So the breadcrumbs which shows Introduction as parent is incorrect.
![vuepress](https://user-images.githubusercontent.com/11680315/81523045-4617c200-936a-11ea-9eee-968b09da985d.png)

**Comment by Mister-Hope** [comment]
> Taking the example from the image below. Here is a demo example
> ```
> .
> ├── guide
> │   ├── basic-config.md
> │   ├── directory-structure.md
> │   ├── install.md
> │   └── README.md
> └ ......
> ```
> Here the file README.md contains the `Introduction`. The file directory-structure.md is in parallel and technically does not reside inside README.md. So the breadcrumbs which shows Introduction as parent is incorrect.
> ![vuepress](https://user-images.githubusercontent.com/11680315/81523045-4617c200-936a-11ea-9eee-968b09da985d.png)
> 

Can you think out a way to express the structure in config? I can make the breadcrumb handle it.

If I am writing this, I will make readme.md as intro.md and only place a catelog of links in readme. And I will not contain readme.md in navbar or sidebar, while using /guide/intro.html instead.

The breadcrumb needs every folder, or from content, as a class of docs, have a catelog as their parent folder.

**Comment by e0lithic** [comment]
Yeah that sounds better using Readme.md as a catalog would better explain the purpose. Thanks

**Comment by e0lithic** [comment]
I think another killer feature could be export the documentation to PDF. In general people do need to get the pdf version of the documentation sooner or later.

**Comment by Mister-Hope** [comment]
> I think another killer feature could be exporting the documentation to PDF. In general people do need to get the pdf version of the documentation sooner or later.

I think that 's not my job 🧐 vuepress is focused on providing online documentation.

**Comment by e0lithic** [comment]
> > I think another killer feature could be exporting the documentation to PDF. In general people do need to get the pdf version of the documentation sooner or later.
> 
> I think that 's not my job  vuepress is focused on providing online documentation.
> 
> Maybe you can look for some tools in npm or other places.

Totally Agreed. I don't have a need for it as well. I was just exploring possible avenues of enhancement. There are a few plugins to achieve it , so no need indeed.

**Comment by e0lithic** [comment]
Hi, I was wondering if a command like `npx vuepress eject` should also be supported for this custom theme. Something like this is required in order to integrate authorization logins for documention using service provides like okta s mentioned [here](https://developer.okta.com/blog/2019/06/14/add-authentication-and-personalization-to-vuepress). If there are any alternate ways to achive the same, kindly share.

**Comment by Mister-Hope** [comment]
> Hi, I was wondering if a command like `npx vuepress eject` should also be supported for this custom theme. Something like this might be  required in order to integrate authorization logins for documention using service provides like okta as mentioned [here](https://developer.okta.com/blog/2019/06/14/add-authentication-and-personalization-to-vuepress).

Sorry for the late reply, I probably missed it. Sure, I will add it in 0.6.0

**Comment by Mister-Hope** [comment]
It's been add experiemental in 0.6.1, just use `vuepress eject-hope`. I have to change my theme structure to make it work. Since vuepress hot loading is not working, it's hard for me to debug. I will leave this project until hot loading is fixed.

**Comment by Mister-Hope** [comment]
> Configurable font size and typography for better utilization of space, for eg : https://nuxtjs.org/guide
Besides, I rewrite part of my style files with `em` so it's possible for you to change font size by writing like

```stylus
// index.styl
html
  font-size 18px

.breadcrumb
  font-size 16px
```

And I am not going to add a option to the `themeConfig` since it's only about style.

**Comment by Mister-Hope** [comment]
> The default search could be extended using flexsearch to include full text search without the need for algolia for private documentation as is being done by vuepress-plugin-fulltext-search

And since my theme has nothing to improve with the vuepress-plugin-fulltext-search plugin, I am not going to bundle it. You can simply use it in your vuepress config.

**Comment by Mister-Hope** [comment]
And about the pdf export feature, I have a look at [vuepress-plugin-export](https://github.com/ulivz/vuepress-plugin-export#readme) and [@snowdog/vuepress-plugin-pdf-export](https://github.com/SnowdogApps/vuepress-plugin-pdf-export)

They are all on W.I.P. , while the first one is abandoned by it's owner, the one who contribute most to vuepress. So I guess there is not a good solution now. Also I don't think I need this feature, so I am not gonna work on it. Maybe you can fork these projects and improve them if you want this feature.

**Comment by Mister-Hope** [comment]
Am I missing anything?

**Comment by Mister-Hope** [comment]
The `vuepress eject-hope` command is working well on v0.6.3.

Thanks again for your advice, if you have more, feel free to open another issue.
